### ISSUE TITLE
Particle symmetrization: what it is, what is implemented where, and what is missing

### ISSUE THREAD

**Issue by Zeyna777** [ComPWA/ampform-dpd#236]
An overview issue: what symmetrization means for an amplitude model, which parts of it the ComPWA stack currently supports, and where the gaps are. It collects one concrete defect (gap 1 below) that is worth splitting off once confirmed.

## What symmetrization is

An isobar model is a coherent sum over decay chains, and by default every chain gets its own free couplings. A symmetry says that some of those chains are **not** independent:

1. partner chains **share one coupling** instead of carrying two, and
2. the **relative sign** between them is fixed by quantum numbers, not fitted.

If the symmetry maps a chain onto *itself*, the relation becomes $H = s H$, which for $s = -1$ is a **selection rule** forcing that coupling to vanish.

Two symmetries have this shape.

**Identical particles.** The exchange operator $P_{ij}$ swaps everything about two particles (momentum and spin). Since $P_{ij}^2 = 1$, its eigenvalue is $\pm 1$, and spin-statistics fixes it: $+1$ for identical bosons (Bose symmetrization), $-1$ for identical fermions (Fermi antisymmetrization). So a final state with identical particles requires

$$
A(\ldots i \ldots j \ldots) = \pm A(\ldots j \ldots i \ldots)
$$

**Charge conjugation.** $C$ relabels particles as antiparticles without touching a single momentum. If the initial state is a $C$ eigenstate and $C$ maps the final state onto itself as a set, then $A(f) = C_0 c_f^* A(\bar{f})$, which again ties chain pairs together.

Both produce the same structure, because the relabelling has to be reconciled with whichever order the code writes a vertex's two children in. That reconciliation costs a **vertex exchange phase**

$$
\begin{aligned}
\eta^{LS} &= (-1)^{\ell + s_i + s_j - S}, \\
\eta^{\mathrm{hel}} &= (-1)^{J_R - s_i - s_j}.
\end{aligned}
$$

and the tie sign is that phase times a prefactor which is the only thing distinguishing the two cases:

$$
\begin{aligned}
\text{identical particles:} & s = (-1)^{2s} \prod_v \eta_v, \\
\text{charge conjugation:} & s = C_0 \prod_a C_a \prod_v \eta_v.
\end{aligned}
$$

Observable consequence in both cases: the intensity is invariant under the induced permutation of the Mandelstam variables, i.e. the Dalitz plot is mirror-symmetric.

## What is implemented where

| | selection rules | chain permutations | tied couplings | correct sign |
|---|:-:|:-:|:-:|:-:|
| **qrules** | ✅ boson + fermion, C, G, parity | ✅ | n/a | n/a |
| **ampform** | (from qrules) | ✅ identical particles | ✅ | ✅ identical particles<br>❌ fermionic $-1$<br>❌ no C |
| **ampform-dpd** | (from qrules) | ✅ identical particles | ✅ implicit | ❌ identical particles<br>❌ fermionic $-1$<br>🚧 C on `c-parity-sign` |

### qrules

- [`conservation_rules.identical_particle_symmetrization`](https://qrules.readthedocs.io/0.10.13/api/qrules.conservation_rules.html#qrules.conservation_rules.identical_particle_symmetrization): selection rule with **both** branches (bosons need mother parity $+1$, fermions $-1$). It fires only when the identical daughters have equal spin projections, which is exactly the regime where it is rigorous: equal projections force even $S$, so "$`L+S`$ even" collapses to the mother-parity test.
- [`conservation_rules.c_parity_conservation`](https://qrules.readthedocs.io/0.10.13/api/qrules.conservation_rules.html#qrules.conservation_rules.c_parity_conservation), [`g_parity_conservation`](https://qrules.readthedocs.io/0.10.13/api/qrules.conservation_rules.html#qrules.conservation_rules.g_parity_conservation), [`parity_conservation_helicity`](https://qrules.readthedocs.io/0.10.13/api/qrules.conservation_rules.html#qrules.conservation_rules.parity_conservation_helicity)
- [`combinatorics.perform_external_edge_identical_particle_combinatorics`](https://qrules.readthedocs.io/0.10.13/api/qrules.combinatorics.html#qrules.combinatorics.perform_external_edge_identical_particle_combinatorics) regenerates the permutations that qrules collapses, since it reasons about quantum states rather than kinematics

✅ selection rules &nbsp;&nbsp; ❌ no amplitude phases; by design, qrules says *whether* a chain is allowed, never with what sign it enters

### ampform

- [`helicity.decay.perform_combinatorics`](https://ampform.readthedocs.io/0.16.1/api/ampform.helicity.decay.html#ampform.helicity.decay.perform_combinatorics) wraps the qrules function
- [`HelicityAmplitudeBuilder.__formulate_topology_amplitude`](https://github.com/ComPWA/ampform/blob/0.16.1/src/ampform/helicity/__init__.py#L436) sums the permutations into **one** amplitude symbol with shared parameters; that is the symmetrization
- [`kinematics.HelicityAdapter.permutate_registered_topologies`](https://ampform.readthedocs.io/0.16.1/api/ampform.kinematics.html#ampform.kinematics.HelicityAdapter.permutate_registered_topologies) registers the permuted topologies so the kinematic variables exist
- [`docs/usage/helicity/symmetrization.ipynb`](https://ampform.readthedocs.io/0.16.1/usage/helicity/symmetrization.html)
- the sign flips that notebook shows for $J/\psi \to \gamma \pi^0 \pi^0$ come from the qrules [`parity_prefactor`](https://qrules.readthedocs.io/0.10.13/api/qrules.quantum_numbers.html#qrules.quantum_numbers.NodeQuantumNumbers.parity_prefactor) node QN, not from exchange symmetry

✅ Bose symmetrization &nbsp;&nbsp; ❌ no fermionic $-1$ &nbsp;&nbsp; ❌ no C symmetrization

### ampform-dpd

- [`adapter.qrules.permute_equal_final_states`](https://ampform-dpd.readthedocs.io/0.4.3/api/ampform_dpd.adapter.qrules.html#ampform_dpd.adapter.qrules.permute_equal_final_states): the DPD analogue of ampform's combinatorics; only user in this repo is [`docs/xib2pkk.ipynb`](https://ampform-dpd.readthedocs.io/0.4.3/xib2pkk.html)
- symmetrization itself is **implicit**: [`_get_coefficient_base`](https://github.com/ComPWA/ampform-dpd/blob/0.4.3/src/ampform_dpd/__init__.py#L374) / [`_get_coupling_base`](https://github.com/ComPWA/ampform-dpd/blob/0.4.3/src/ampform_dpd/__init__.py#L389) key couplings on the resonance name only, never on the subsystem, so permuted chains automatically share one coupling and are summed with $+1$
- [`_order_decay_products`](https://github.com/ComPWA/ampform-dpd/blob/0.4.3/src/ampform_dpd/__init__.py#L397) enforces the cyclic pair ordering $(23)1, (31)2, (12)3$, the prerequisite for a well-defined $\eta$
- [`cparity.py`](https://github.com/ComPWA/ampform-dpd/blob/668886b05778c2f74dbd9edba50910ea0b6842e8/src/ampform_dpd/cparity.py) on the `c-parity-sign` branch (#204) implements the charge-conjugation case in full: [`get_exchange_phase`](https://github.com/ComPWA/ampform-dpd/blob/668886b05778c2f74dbd9edba50910ea0b6842e8/src/ampform_dpd/cparity.py#L199), [`get_conjugate_coupling_sign`](https://github.com/ComPWA/ampform-dpd/blob/668886b05778c2f74dbd9edba50910ea0b6842e8/src/ampform_dpd/cparity.py#L159), [`get_c_forbidden_chains`](https://github.com/ComPWA/ampform-dpd/blob/668886b05778c2f74dbd9edba50910ea0b6842e8/src/ampform_dpd/cparity.py#L273), [`symmetrize_conjugate_couplings`](https://github.com/ComPWA/ampform-dpd/blob/668886b05778c2f74dbd9edba50910ea0b6842e8/src/ampform_dpd/cparity.py#L323)

## Gaps

**1. The identical-particle tie uses $+1$ instead of $\eta$.** The shared-coupling mechanism supplies no exchange phase, so it is correct only when $\eta = +1$ and builds the *antisymmetric* combination when $\eta = -1$.

<details>
<summary>Evidence</summary>

$D^+ \to \pi^+ \pi^+ \pi^-$. States 1 and 2 are identical, so the exchange is $\sigma_1 \leftrightarrow \sigma_2$ and the Dalitz diagonal is where it acts trivially. A Bose-symmetric amplitude is generically non-zero there; an antisymmetric one has an exact node along the whole diagonal.

| framework | resonance | $\ell$ | $\eta$ | intensity on $\sigma_1 = \sigma_2$ |
|---|---|:-:|:-:|---|
| **ampform-dpd** | $\rho(770)^0$ | 1 | $-1$ | $4.4 \times 10^{-15}$ vs max $9.6 \times 10^{-2}$ → **exact node** |
| **ampform-dpd** | $f_2(1270)$ | 2 | $+1$ | $1.09 \times 10^{-1}$ vs max $1.14 \times 10^{-1}$ → no node |
| ampform | $\rho(770)^0$ | 1 | n/a | $\langle I \rangle$ near/away diagonal = $2.99$ → enhancement |
| ampform | $f_2(1270)$ | 2 | n/a | $2.91$ → enhancement |

Mirror asymmetry was $\sim 3 \times 10^{-7}$ in every ampform-dpd case, so the axis is right. Same outcome with `min_ls=True` and `min_ls=False`, so it is not a coupling-basis artifact.

Analytic cross-check with Zemach spin factors $Z_A = (p_1-p_3) \cdot p_2$, $Z_B = (p_2-p_3) \cdot p_1$:

- $Z_A + Z_B \propto m_D^2 + 3 m_\pi^2 - 3 \sigma$; one isolated zero
- $Z_A - Z_B = p_3 \cdot (p_1-p_2)$; zero on the **whole** diagonal

ampform-dpd reproduces the second, ampform the first.

**Origin.** With identical states 1 and 2, the exchange maps chain $(31)2$ onto children $(3,2)$, but the code writes spectator-1 chains as $(2,3)$. One reversal on the decay vertex, costing $\eta$. The production vertex $0 \to R k$ always matches, exactly as documented for the C case on `c-parity-sign`.

**Likely impact on [`docs/xib2pkk.ipynb`](https://ampform-dpd.readthedocs.io/0.4.3/xib2pkk.html).** Parity conservation at $R \to p K^-$ gives $(-1)^\ell = -P_R$, hence $\eta^{LS} = -P_R$:

| resonance | $J^P$ | $\eta^{LS}$ | $\eta^{\mathrm{hel}}$ |
|---|:-:|:-:|:-:|
| $\Lambda(1405)$ | $\frac{1}{2}^-$ | $+1$ | $+1$ |
| $\Lambda(1520)$ | $\frac{3}{2}^-$ | $+1$ | **$-1$** |
| $\Lambda(1670)$ | $\frac{1}{2}^-$ | $+1$ | $+1$ |
| $\Sigma(1385)$ | $\frac{3}{2}^+$ | **$-1$** | **$-1$** |
| $\Sigma(1775)$ | $\frac{5}{2}^-$ | $+1$ | $+1$ |
| $\Sigma(1915)$ | $\frac{5}{2}^+$ | **$-1$** | $+1$ |

**This table is a prediction from the closed form, not a measurement**; that reaction did not converge in qrules for me, so the notebook was not run end to end.

</details>

<details>
<summary>Reproduction</summary>

```python
import logging, warnings
logging.getLogger("qrules.transition").setLevel(logging.ERROR)
warnings.simplefilter("ignore")
import jax.numpy as jnp, qrules
from tensorwaves.data.transform import SympyDataTransformer
from ampform_dpd import DalitzPlotDecompositionBuilder
from ampform_dpd.adapter.qrules import (
    normalize_state_ids, permute_equal_final_states, to_three_body_decay
)
from ampform_dpd.dynamics.builder import formulate_breit_wigner_with_form_factor
from ampform_dpd.io import cached

for res in ["rho(770)0", "f(2)(1270)"]:
    r = qrules.generate_transitions(
        initial_state="D+", final_state=["pi+", "pi+", "pi-"],
        allowed_intermediate_particles=[res], formalism="canonical-helicity")
    decay = to_three_body_decay(
        permute_equal_final_states(normalize_state_ids(r)).transitions, min_ls=True)
    b = DalitzPlotDecompositionBuilder(decay, min_ls=True)
    for c in b.decay.chains:
        b.dynamics_choices.register_builder(c, formulate_breit_wigner_with_form_factor)
    model = b.formulate(reference_subsystem=1)

    m = {k: float(v) for k, v in model.masses.items()}
    m0, m1, m2, m3 = (m[s] for s in sorted(m, key=str))
    lo, hi = (m2 + m3) ** 2, (m0 - m1) ** 2
    N = 401
    X, Y = jnp.meshgrid(jnp.linspace(lo, hi, N), jnp.linspace(lo, hi, N))
    s3, s3_expr = list(model.invariants.items())[2]
    defs = dict(model.variables)
    defs[s3] = s3_expr
    defs = {k: v.xreplace(defs).xreplace(model.masses) for k, v in defs.items()}
    data = {"sigma1": X, "sigma2": Y}
    data.update(SympyDataTransformer.from_sympy(defs, backend="jax")(data))
    func = cached.lambdify(
        cached.xreplace(cached.unfold(model), dict(model.parameter_defaults)), backend="jax")
    I = func(data)

    diag = jnp.array([I[i, i] for i in range(N)])
    diag = jnp.where(jnp.isfinite(diag), diag, jnp.nan)
    print(f"{res:12s} diagonal/max = {float(jnp.nanmax(diag) / jnp.nanmax(I)):.4f}")
```

```
rho(770)0    diagonal/max = 0.0000
f(2)(1270)   diagonal/max = 0.9566
```

</details>

**2. No fermionic $\epsilon = -1$.** Neither ampform nor ampform-dpd can express antisymmetrization; permuted chains are simply added. Rare in three-body hadronic final states (you need two identical baryons), but there is currently no way to write it.

**3. [`permute_equal_final_states`](https://ampform-dpd.readthedocs.io/0.4.3/api/ampform_dpd.adapter.qrules.html#ampform_dpd.adapter.qrules.permute_equal_final_states) covers only part of $S_3$.** It builds permutations from `itertools.combinations(equal_state_ids, 2)`, i.e. identity plus transpositions, covering 4 of the 6 elements for **three** identical final-state particles. Those 4 already reach all three subsystems and [`_order_decay_products`](https://github.com/ComPWA/ampform-dpd/blob/0.4.3/src/ampform_dpd/__init__.py#L397) normalizes the pair ordering afterwards, so it is probably harmless, but untested.

**4. Nothing documents that symmetrization happens.** In ampform-dpd it is a side effect of the coupling-naming convention. Keying couplings per subsystem would silently break it.

## Notes for whoever picks this up

- [`get_exchange_phase(chain, basis)`](https://github.com/ComPWA/ampform-dpd/blob/668886b05778c2f74dbd9edba50910ea0b6842e8/src/ampform_dpd/cparity.py#L199) from [`cparity.py`](https://github.com/ComPWA/ampform-dpd/blob/668886b05778c2f74dbd9edba50910ea0b6842e8/src/ampform_dpd/cparity.py) is already the function gap 1 needs. An identical-particle counterpart to [`symmetrize_conjugate_couplings`](https://github.com/ComPWA/ampform-dpd/blob/668886b05778c2f74dbd9edba50910ea0b6842e8/src/ampform_dpd/cparity.py#L323) would be the same code with $C_0 \prod_a C_a$ replaced by $(-1)^{2s}$, closing gaps 1 and 2 together.
- For regression tests use the **diagonal**, not the mirror asymmetry: both $|A+B|^2$ and $|A-B|^2$ are mirror-symmetric (they are the $+$/$`-`$ eigenstates of the tie), so the mirror check cannot see this sign. The same caveat applies to the C-parity tie in #204.

Related: #204 (C-parity sign, same $\eta$), #108 (support for equal final states), #16.
